frontend/src/app/components/crm/leads/leads.component.ts

```diff
@@ -149,6 +155,47 @@ export class LeadsComponent implements AfterViewInit {
     }
     return `${this.selection.isSelected(row) ? 'deselect' : 'select'} row ${row.leadId + 1}`;
   }
+  /*
+   * This method listens to the contact leads button
+   */
+  contactLead(){
+    let recepientsString : string = "";
+
+    // when no leads are selected
+    if(this.selection.selected.length == 0){
+      alert("Please select at least one lead.");
+      return;
+    }
+
+    let recepients: Array<string> = [];
+
+    for (let i = 0; i < this.selection.selected.length; i++) {
+      if(this.selection.selected[i].columnData.EMAIL == null ){
+         //do nothing
+      }else{
+        recepients.push(this.selection.selected[i].columnData.EMAIL.toString());
+      }
+    }
+
+    // incase all the selected leads do not have email address
+    if(recepients.length == 0){
+    alert("Please select at least one lead with an email address.");
+          return;
+    }
+
+    for(let j = 0; j < recepients.length; j++){
```

Comment:
You can use [`Array.reduce`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/reduce) here.

Reply:
okay, used Array.join instead